### ISSUE TITLE
Linting Error: AttributeError: 'NoneType' object has no attribute 'strip'

### ISSUE THREAD

**Issue by dwallace0723** [sqlfluff/sqlfluff#780]
We get this error for ~10-20 different dbt models within our repository. The exception says to report it as an issue.

**Example Compiled SQL:**

```sql
select
    users.dbt_scd_id,
    users.valid_date_local,
    users._id,
    users.emails,
    users.created_at,
    users.updated_by,
    users.created_by,
    users.profile:name::string as name,
    users.profile:status::string as status,
    users.profile,
    users.profile:address::string as address,
    users.profile:chain::string as chain,
    users.medical_dispensary
from PUBLIC.users_hist__dbt as users
where users.profile:type::varchar = 'Dispensary'
```

**sqlfluff output:**
```py
WARNING    Unable to lint models/content/dispensaries_hist.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1252, in lint_path
    self.lint_string(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1045, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 873, in parse_string
    parsed: Optional[BaseSegment] = parser.parse(tokens, recurse=recurse)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/parser.py", line 30, in parse
    parsed = root_segment.parse(parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 21, in parse
    new_self = super().parse(parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 759, in parse
    m = self.parse_grammar.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 430, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 430, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/delimited.py", line 97, in match
    ) = self._bracket_sensitive_look_ahead_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 550, in _bracket_sensitive_look_ahead_match
    pre, match, matcher = cls._look_ahead_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 381, in _look_ahead_match
    mat, m = cls._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 133, in _match_once
    available_options, _ = self._prune_options(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 67, in _prune_options
    if elem.strip():
AttributeError: 'NoneType' object has no attribute 'strip'
```

## Version
`0.4.0`

## Configuration
```
[sqlfluff]
dialect = snowflake
templater = dbt
# dbt templating does not keep trailing new lines (L009)
exclude_rules = L009
```

**Comment by alanmcruickshank** [comment]
Thanks @dwallace0723 - I've just tested your compiled query and I'm having trouble replicated the error on my side.

Could you run sqlfluff in one of the more verbose logging modes (by adding `-vvvvv` to the command)? The output will be *very long* so you don't need to paste all of it. The useful lines will be very near the end, maybe post the last 100 lines or so.

The error is coming from one of the hueristics used during parsing - somehow a `None` value is getting in there, which should be impossible - but is evidently happening somehow.

**Comment by dwallace0723** [comment]
@alanmcruickshank sure thing! Going to actually use a more succinct query that sqlfluff is choking on as an example moving forward. We've updated a few configurations since I've first posted this, so I will post those as well:

**Pre-Compiled SQL:**
```sql
{{
  config(
    materialized = 'incremental',
    unique_key='surrogate_pk',
    cluster_by=['valid_date_local'],
    automatic_clustering=true
    )
}}

select
    {#- Attributes #}
    products.product_id,
    products.name,
    products.brand_name,
    dispensaries.state,
    products.category,
    products.subcategory,
    count(1) over (partition by products.duplicated_product_id, products.valid_date_local) as num_connections,
    avg(products.score) over (partition by products.duplicated_product_id, products.valid_date_local) as avg_score,
    products.description,
    iff(lower(products.name) like '%enter%', true, false) as is_enter_strain,

    {#- Metadata #}
    products.surrogate_pk,
    products.snapshot_id,
    products.valid_date_local,
    products.dbt_valid_from_local,
    products.dbt_valid_to_local,
    products._fivetran_synced,
    products._fivetran_deleted
from {{ ref('products_daily_fanout') }} as products
inner join {{ ref('dispensaries_history') }} as dispensaries
on products.dispensary_id = dispensaries.dispensary_id
and products.valid_date_local = dispensaries.valid_date_local
where dispensaries.profile:stateLibrary::varchar is not null
and not products._fivetran_deleted
{% if is_incremental() -%}
and products.valid_date_local >= (select max(valid_date_local) from {{ this }})
{% endif -%}
```

**Compiled SQL:**
```sql
select
    products.product_id,
    products.name,
    products.brand_name,
    dispensaries.state,
    products.category,
    products.subcategory,
    count(1) over (partition by products.duplicated_product_id, products.valid_date_local) as num_connections,
    avg(products.score) over (partition by products.duplicated_product_id, products.valid_date_local) as avg_score,
    products.description,
    iff(lower(products.name) like '%enter%', true, false) as is_enter_strain,
    products.surrogate_pk,
    products.snapshot_id,
    products.valid_date_local,
    products.dbt_valid_from_local,
    products.dbt_valid_to_local,
    products._fivetran_synced,
    products._fivetran_deleted
from ANALYTICS_DWALLACE.PUBLIC.products_daily_fanout as products
inner join ANALYTICS_DWALLACE.PUBLIC.dispensaries_history as dispensaries
on products.dispensary_id = dispensaries.dispensary_id
and products.valid_date_local = dispensaries.valid_date_local
where dispensaries.profile:stateLibrary::varchar is not null
and not products._fivetran_deleted
and products.valid_date_local >= (select max(valid_date_local) from ANALYTICS_DWALLACE.PUBLIC.products_library_history)
```

**Command:**
```sh
$ sqlfluff parse -vvvvv models/content/products/products_library_history.sql
```

**Output:**
[sqlfluff_output.txt](https://github.com/sqlfluff/sqlfluff/files/6013228/sqlfluff_output.txt)

**Traceback:**
```
Traceback (most recent call last):
  File "/Users/dwall/.pyenv/versions/dbt-dutchie/bin/sqlfluff", line 8, in <module>
    sys.exit(cli())
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/cli/commands.py", line 572, in parse
    for parsed_string in result:
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1309, in parse_path
    yield self.parse_string(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 873, in parse_string
    parsed: Optional[BaseSegment] = parser.parse(tokens, recurse=recurse)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/parser.py", line 30, in parse
    parsed = root_segment.parse(parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 825, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 270, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 759, in parse
    m = self.parse_grammar.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 430, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 112, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 720, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/segments/base.py", line 430, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/delimited.py", line 97, in match
    ) = self._bracket_sensitive_look_ahead_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 550, in _bracket_sensitive_look_ahead_match
    pre, match, matcher = cls._look_ahead_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 381, in _look_ahead_match
    mat, m = cls._longest_trimmed_match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/base.py", line 202, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/match_wrapper.py", line 56, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 197, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 133, in _match_once
    available_options, _ = self._prune_options(
  File "/Users/dwall/.pyenv/versions/3.8.6/envs/dbt-dutchie/lib/python3.8/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 67, in _prune_options
    if elem.strip():
AttributeError: 'NoneType' object has no attribute 'strip'
```

**Comment by alanmcruickshank** [comment]
Ok this was a tough one but a great bug report. In the process of trying to recreate this error I flushed out a few other bugs in the process. I'll put more detail in the PR but I think I've solved this one.
